vcr/cassette.py

```diff
@@ -309,7 +309,7 @@ def find_requests_with_most_matches(self, request):
         best_matches.sort(key=lambda t: t[0], reverse=True)
         # Get the first best matches (multiple if equal matches)
         final_best_matches = []
-        previous_nb_success = best_matches[0][0]
```

Comment:
If the cassette is empty, it means that `best_matches` is empty too. Just add a return statement if we are in this case.